src/python/ensembl/io/genomio/events_loader.py

```diff
@@ -0,0 +1,166 @@
+
+from ensembl.database import DBConnection
+from ensembl.core.models import MappingSession, StableIdEvent
+
+
+@dataclass
+class IdEvent:
+    """Simple representation for the events from the input file"""
+    from_id: str
+    to_id: str
+    release: str
+    release_date: str
+
+
+class MapSession:
+    """Simple mapping_sessions representation from the input file"""
+    def __init__(self, release: str, release_date: str) -> None:
+        self.release = release
+        self.release_date = release_date
+        self.events: List[IdEvent] = []
+
+    def add_event(self, event: IdEvent) -> None:
+        """Add an event to this mapping_session"""
+        self.events.append(event)
+
+
+def load_events(input_file: Path) -> List[IdEvent]:
+    """Load events from input file.
+    Expected tab file columns: old_id, new_id, event_name, release, release_date
+    
+    """
+    events: List[IdEvent] = []
+
+    with input_file.open("r") as events_fh:
+        for line in events_fh:
+            line.strip()
+            if line == "":
+                continue
+            (from_id, to_id, _, release, release_date) = line.split("\t")
+            event = IdEvent(from_id=from_id, to_id=to_id, release=release, release_date=release_date)
+            events.append(event)
+    return events
+
+
+def write_events(session: Session, events: List[IdEvent], update: bool = False) -> None:
+    """Insert the events in the core database.
+    A mapping session is created for each different 'release'.
+    
+    """
+    # First, create mapping_sessions based on the release
+    mappings: Dict[str, MapSession] = {}
+    for event in events:
+        release = event.release
+
+        if release in mappings:
+            mappings[release].add_event(event)
+        else:
+            mappings[release] = MapSession(release, event.release_date)
+
+    # Then, add the mapping, and the events for this mapping
+    for release, mapping in mappings.items():
+        print(f"Adding mapping for release {release} ({len(mapping.events)} events)")
+        if update:
+            map_session = MappingSession(new_release=mapping.release, created=mapping.release_date)
+            session.add(map_session)
+            session.flush()
+            session.refresh(map_session)
+            for event in mapping.events:
+                id_event = StableIdEvent(
+                    mapping_session_id=map_session.mapping_session_id,
+                    old_stable_id=event.from_id,
+                    new_stable_id=event.to_id,
+                    id_type="gene",
+                    old_version=1,
+                    new_version=1,
+                )
+                session.add(id_event)
+            session.commit()
+
+
+class InputSchema(argschema.ArgSchema):
+    """Input arguments expected by this script."""
+
+    # Server parameters
+    host = argschema.fields.String(
+        required=True, metadata={"description": "Host to the server with EnsEMBL databases"}
```

Comment:
I have also seen previously the option to write this as:
```
metadata={"required": True, "description": "Host to the server with EnsEMBL databases"}
```
In my opinion, this is something that should be fixed in `argschema` so `description` can go again outside of `metadata`, but I'm afraid it won't happen any time soon.

Reply:
It doesn't seem to work properly when it is in metadata (and the only way to tell is by running the script).